### ISSUE TITLE
Database readonly and Split Read/Write

### ISSUE THREAD

**Issue by Sungodragon** [ExpressionEngine/ExpressionEngine#508]
**Is your feature request related to a problem? Please describe.**
Problem Horizontal Scaling especially different continents.

**Describe the solution you'd like**

Topology One Master and Multiple Slaves (Read-Only) [In All top clouds, Mysql Read-only slaves are possible]
Read Only slaves can be geographically in a different continent

ExpressionEngine should have the ability to add Master Database, Read-Only Slaves.
Ability to add multiple slaves for even higher availability.

ExpressionEngine will not take care of the replication but should be able to utilize the slaves for read-only.

**Describe alternatives you've considered**
N/A

**Teachability, Documentation, Adoption, Migration Strategy**
N/A

- [ X] I am capable and would like to work on the implementation of this feature if it is considered.

**Comment by Sungodragon** [comment]
Any plans for this in EE6 or will need to wait for EE7?

**Comment by Sungodragon** [comment]
Hi is there any alternative to achieve this? i.e. outside the scope of EE which can be implemented.

Will a forum post help better or on Discord?
Just for reference, this kind of feature is available in WordPress, Craftcms to name a few.
This feature will be a Game Changer for EE, especially for Global Companies and Enterprises.

**Comment by intoeetive** [comment]
Yes - if you post on EE [Forums](https://expressionengine.com/forums) or [Slack](https://expressionengine.com/blog/join-us-in-slack), there's a chance someone in the community could have done it before and can help.

**Comment by Sungodragon** [comment]
I have asked this in Slack but no response. So I have following questions

1) Do we need this functionality in Core EE to achieve this?
  
2) Can one develop a plugin to achieve this?

**Comment by intoeetive** [comment]
Well, so far you have been the only person to request this. If no one is also requesting this in Slack... well, IDK

I don't think this can be achieved with a plugin. The database function are loaded early, so this needs to be core component

**Comment by compareshow** [comment]
This feature to split the database is very important. This is definitely a factor nowadays to choose a CMS. 

This needs to be either in 'core' or a 'Pro' feature at least as this is a important requirement for serious business (who will definitely opt for Pro.)
@intoeetive @TomJaeger 

All the best .
